python/valis/routes/target.py

```diff
@@ -181,7 +182,18 @@ async def get_spectrum(self, sdss_id: Annotated[int, Path(title="The sdss_id of
                 response_model_exclude_unset=True, response_model_exclude_none=True)
     async def get_catalogs(self, sdss_id: int = Path(title="The sdss_id of the target to get", example=23326)):
         """ Return catalog information for a given sdss_id """
-        return get_catalog_sources(sdss_id).dicts().iterator()
+        sdss_id_data = get_catalog_sources(sdss_id).dicts()
+
+        parent_catalogs: list[dict[str, Any]] = []
+        for i in sdss_id_data:
+            cat_data = get_parent_catalogs(i['catalogid'])
+            if cat_data.count() > 0:
+                parent_catalogs.append(cat_data.dicts()[0])
+            else:
+                parent_catalogs.append({})
+
+        return [CatalogResponse(**s_data, parent_catalogs=parent_catalogs[idx])
+                for idx, s_data in enumerate(sdss_id_data)]
```

Comment:
Is there a way to simplify this any?  we set the endpoint `response_model=List[CatalogResponse]`, so in principle we can just return a list or iterator of dict information and it will do the model serialization for us.  Is there a way to take peewee's `.dicts().iterator()` and yield a new merged iterator?

Reply:
I think the best we can do is to create a new response dictionary as a dictionary comprehension and then let FastAPI convert it to the model when the route returns. But the ugly `for idx, s_data ...` will still be there, I think and it won't be a generator. But the number of items returned by this query is always very small for performance-wise it's unlikely to affect much.

Reply:
Or basically this, which works

```python
response = ({**s_data, 'parent_catalogs': parent_catalogs[idx]}
                    for idx, s_data in enumerate(sdss_id_data))
return response
```

Reply:
This new `response` looks cleaner to me.